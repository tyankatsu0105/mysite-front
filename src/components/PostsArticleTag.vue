<template>
	<div
		class="PostsArticleTag"
	>
		<ul
			class="PostsArticleTag-List"
		>
			<li
				v-for="tag in node"
				:key="tag.sys.id"
				class="PostsArticleTag-Item"
			>
				#{{ tag.fields.title }}
			</li>
		</ul>
	</div>
</template>

<script>
export default {
  props: {
    node: {
      type: Array,
      default: () => []
    }
  }
};
</script>

<style scoped lang='scss'>
.PostsArticleTag {
  &-List {
    display: flex;
    flex-wrap: wrap;
  }
  &-Item {
    padding: 4px 8px;
    margin-right: 4px;
    margin-bottom: 4px;
    font-size: 0.8rem;
    background-color: lighten($color-primary, 10%);
    border-radius: 5px;
  }
}
</style>
