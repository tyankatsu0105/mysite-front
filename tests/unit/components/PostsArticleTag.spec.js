import { shallowMount } from "@vue/test-utils";
import PostsArticleTag from "@/components/PostsArticleTag.vue";

const propsData = {
  node: [
    {
      title: "Gridsome",
      id: "5HQR4kLzgc0IAeKoM0EwgI"
    },
    {
      title: "FrontEnd",
      id: "3uo6xqFMfuOaosgiOemwCa"
    }
  ]
};

describe("PostsArticleTag", () => {
  it("render", () => {
    const wrapper = shallowMount(PostsArticleTag, {
      propsData,
      stubs: ["g-link", "g-image", "font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
