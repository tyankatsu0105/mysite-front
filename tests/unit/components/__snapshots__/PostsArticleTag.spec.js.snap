// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsArticleTag render 1`] = `
<div class="PostsArticleTag">
  <ul class="PostsArticleTag-List">
    <li class="PostsArticleTag-Item">
      #Gridsome
    </li>
    <li class="PostsArticleTag-Item">
      #FrontEnd
    </li>
  </ul>
</div>
`;
